plugins/test/dynamic_test.cc

```diff
@@ -287,12 +353,44 @@ void DynamicTest::BenchHttpHandlers(benchmark::State& state) {
     response_headers = *headers;
   }
   std::vector<std::string> request_body_chunks;
-  for (const auto& request_body : cfg_.request_body()) {
-    request_body_chunks.emplace_back(request_body.input().content());
+  if (cfg_.request_body_size() > 0) {
+    if (cfg_.chunking_plan_case() !=
+        pb::Test::ChunkingPlanCase::CHUNKING_PLAN_NOT_SET) {
+      if (cfg_.request_body_size() > 1) {
+        FAIL() << "Cannot specify chunking_plan with multiple request_body "
+                  "invocations";
+      } else {
+        auto complete_body = ParseBodyInput(cfg_.request_body(0).input());
+        BM_RETURN_IF_ERROR(complete_body.status());
+        request_body_chunks = ChunkBody(*complete_body, cfg_);
+      }
+    } else {
+      for (const auto& request_body : cfg_.request_body()) {
+        auto body_chunk = ParseBodyInput(request_body.input());
+        BM_RETURN_IF_ERROR(body_chunk.status());
+        request_body_chunks.emplace_back(*body_chunk);
+      }
+    }
   }
   std::vector<std::string> response_body_chunks;
-  for (const auto& response_body : cfg_.response_body()) {
-    response_body_chunks.emplace_back(response_body.input().content());
+  if (cfg_.response_body_size() > 0) {
```

Comment:
Can this be factored out into a helper method to reduce duplication with lines 356--373?